### PR TITLE
Fix Issue 19882 - Expected 'undefined identifier' error when type

### PR DESCRIPTION
**Pull request by kubo39**
comparing

### CODE REVIEW

**Comment by kubo39** [comment]
Hm, this fix is wrong...

File: src/dmd/expressionsem.d [hunk 1]
```diff
@@ -5038,7 +5038,11 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         sc2.tinst = null;
         sc2.minst = null;
         sc2.flags |= SCOPE.fullinst;
-        Type t = e.targ.trySemantic(e.loc, sc2);
+        Type t = null;
+        if (e.targ.ty == Tident)
+            t = e.targ.typeSemantic(e.loc, sc2);
```
Comment (kubo39) [inline src/dmd/expressionsem.d]: Ah, set error only if `undefined identifier`.

**Comment by kubo39** [comment]
I'm wrong, it should be compilable.

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
Is this still WIP?

**Comment by kubo39** [comment]
Ah, no more WIP. Thank you!

**Comment by thewilsonator** [comment]
Thx!

**Pull request merged**